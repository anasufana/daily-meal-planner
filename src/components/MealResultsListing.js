import React from 'react';
import MealCard from './MealCard';
import '../css/components/MealResultsListing.css';
import '../css/components/MealCard.css';

const MealResultsListing = (props) => {
  if (props.apiResponse) {
    return (
      <div className="results-container">
        <div className="meal-calories-label">
          <h3 className="h3-calories">Total day calories:
            <span className="calories-amount">{ props.apiResponse.nutrients.calories }</span>
          </h3>
        </div>
        <div className="meal-results-listing">
          <MealCard meal="breakfast" details={props.apiResponse.meals[0]} />
          <MealCard meal="lunch" details={props.apiResponse.meals[1]} />
          <MealCard meal="dinner" details={props.apiResponse.meals[2]} />
        </div>
      </div>
    );
  }
  return '';
};

export default MealResultsListing;
