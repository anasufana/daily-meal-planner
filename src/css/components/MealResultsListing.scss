* {
  margin: 0;
  padding: 0;
}

a {
  text-decoration: none;
}
a:visited,
a:link {
  color: #555;
}
.mealCaloriesLabel {
  text-align: left;
  width: 55%;
  margin: 0 auto;
  padding-top: 20px;
  border-top: 1px solid rgba(0,0,0,.3);
}

.h3Calories {
  display: inline-block;
  padding: 8px 10px;
  background-color: white;
  border-radius: 5px 5px 0 0;
  border-bottom: 3px solid black;
}

.meal-results-listing {
  width: 55%;
  margin: 0 auto;
  padding: 1rem 0 2em 0;
  /* border-top: 1px solid black; */
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-column-gap: 20px;
}

@media screen and (max-width: 550px) {
  .meal-planner-input {
    width: 300px;
  }
}

@media screen and (max-width: 769px) {
  .meal-results-listing {
    width: 75%;
    margin: 0 auto;
    padding: 1rem 0 2em 0;
    /* border-top: 1px solid black; */
    display: grid;
    grid-template-columns: 1fr;
    grid-column-gap: 20px;
  }
  .mealCaloriesLabel {
    width: 75%;
  }
}

@media only screen and (min-width: 769px) and (max-width: 1024px) {
  .meal-results-listing, 
  .mealCaloriesLabel {
    width: 75%;
  }
}

