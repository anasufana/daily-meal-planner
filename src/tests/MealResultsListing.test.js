/* global it, expect, jest */

import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import MealResultsListing from '../components/MealResultsListing';
import apiResponse2 from '../mockAPIresponse/apiResponse';

it('renders correctly', () => {
  const tree = renderer
    .create(<Router>
      <MealResultsListing
        apiResponse={apiResponse2}
        error={false}
        handleMealRequest={jest.fn()}
      />
    </Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
