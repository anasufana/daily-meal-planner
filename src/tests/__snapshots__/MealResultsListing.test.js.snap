// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="results-container"
>
  <a
    className="back-link"
    href="/"
    onClick={[Function]}
  >
    &lt; Go back
  </a>
  <div
    className="meal-calories-label"
  >
    <h3
      className="h3-calories"
    >
      Total day calories:
      <span
        className="calories-amount"
      >
        1695.24
      </span>
    </h3>
  </div>
  <div
    className="meal-results-listing"
  >
    <div
      className="single-column slide-in-1"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <h2
        className="meal-title"
      >
        breakfast
      </h2>
      <div
        className="img-container"
      >
        <img
          alt="Breakfast-recipe"
          className="image-recipe"
          src="https://spoonacular.com/recipeImages/Salmon-Artichoke-Quiche-516473.png"
          title="Breakfast-recipe"
        />
        <div
          className="hover-box"
        >
          <h3
            className="hover-title"
          >
            Time to cook!
          </h3>
          <img
            alt="Time"
            className="img-cooking"
            height="60"
            src="/img/cooking.svg"
            title="Time"
            width="60"
          />
        </div>
      </div>
      <div
        className="meal-details-container"
      >
        <div
          className="meal-details"
        >
          <h3
            className="recipe-title"
          >
            Salmon Artichoke Quiche
             
          </h3>
        </div>
        <div
          className="meal-details clock-div"
        >
          <img
            alt="Time"
            className="img-clock"
            height="20"
            src="/img/clock.svg"
            title="Time"
            width="20"
          />
          <h3
            className="recipe-duration"
          >
            95
            '
          </h3>
        </div>
      </div>
    </div>
    <div
      className="single-column slide-in-2"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <h2
        className="meal-title"
      >
        lunch
      </h2>
      <div
        className="img-container"
      >
        <img
          alt="Breakfast-recipe"
          className="image-recipe"
          src="https://spoonacular.com/recipeImages/Thai-pesto-prawn-noodles-220012.jpg"
          title="Breakfast-recipe"
        />
        <div
          className="hover-box"
        >
          <h3
            className="hover-title"
          >
            Time to cook!
          </h3>
          <img
            alt="Time"
            className="img-cooking"
            height="60"
            src="/img/cooking.svg"
            title="Time"
            width="60"
          />
        </div>
      </div>
      <div
        className="meal-details-container"
      >
        <div
          className="meal-details"
        >
          <h3
            className="recipe-title"
          >
            Thai pesto prawn noodles
             
          </h3>
        </div>
        <div
          className="meal-details clock-div"
        >
          <img
            alt="Time"
            className="img-clock"
            height="20"
            src="/img/clock.svg"
            title="Time"
            width="20"
          />
          <h3
            className="recipe-duration"
          >
            20
            '
          </h3>
        </div>
      </div>
    </div>
    <div
      className="single-column slide-in-3"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <h2
        className="meal-title"
      >
        dinner
      </h2>
      <div
        className="img-container"
      >
        <img
          alt="Breakfast-recipe"
          className="image-recipe"
          src="https://spoonacular.com/recipeImages/mahi-mahi-with-mango-vanilla-coulis-300236.jpeg"
          title="Breakfast-recipe"
        />
        <div
          className="hover-box"
        >
          <h3
            className="hover-title"
          >
            Time to cook!
          </h3>
          <img
            alt="Time"
            className="img-cooking"
            height="60"
            src="/img/cooking.svg"
            title="Time"
            width="60"
          />
        </div>
      </div>
      <div
        className="meal-details-container"
      >
        <div
          className="meal-details"
        >
          <h3
            className="recipe-title"
          >
            Mahi Mahi with Mango-Vanilla Coulis
             
          </h3>
        </div>
        <div
          className="meal-details clock-div"
        >
          <img
            alt="Time"
            className="img-clock"
            height="20"
            src="/img/clock.svg"
            title="Time"
            width="20"
          />
          <h3
            className="recipe-duration"
          >
            30
            '
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
